soc/realtek/ameba/amebad/CMakeLists.txt

```diff
@@ -0,0 +1,34 @@
+# Copyright (c) 2024 Realtek Semiconductor Corp.
+# SPDX-License-Identifier: Apache-2.0
+
+zephyr_include_directories(${ZEPHYR_BASE}/drivers)
+zephyr_include_directories(.)
+
+zephyr_sources(soc.c)
+
+zephyr_linker_sources(SECTIONS boot_section.ld)
+set(SOC_LINKER_SCRIPT ${ZEPHYR_BASE}/include/zephyr/arch/arm/cortex_m/scripts/linker.ld CACHE INTERNAL "")
+
+# west sign
+file(GLOB BIN_FILES "${ZEPHYR_HAL_REALTEK_MODULE_DIR}/zephyr/blobs/${CONFIG_SOC_SERIES}/bin/*.bin")
```

Comment:
this should only ever be done if the user wants it and there is a Kconfig that specifically does not do this, and is meant for **drivers**, as per the blobs page https://docs.zephyrproject.org/latest/contribute/bin_blobs.html#allowed-types

> Hardware agnostic features provided via a proprietary library are not acceptable. For example, a proprietary and hardware agnostic TCP/IP stack distributed as a static archive would be rejected.

I am not seeing what gets added here, it should be one symbol for a specific driver = one blob and should be in code in the `module` folder not `soc` folder since blobs are modules